packages/teams-js/src/internal/communication.ts

```diff
@@ -615,39 +664,52 @@ function handleParentMessage(evt: DOMMessageEvent): void {
   if ('id' in evt.data && typeof evt.data.id === 'number') {
     // Call any associated Communication.callbacks
     const message = evt.data as MessageResponse;
-    const callback = CommunicationPrivate.callbacks[message.id];
-    logger('Received a response from parent for message %i', message.id);
+    const callbackId = retrieveMessageUUIDFromResponse(message);
+    const callback = CommunicationPrivate.callbacks.get(callbackId);
+    logger('Received a response from parent for message %i', callbackId);
     if (callback) {
-      logger('Invoking the registered callback for message %i with arguments %o', message.id, message.args);
+      logger('Invoking the registered callback for message %i with arguments %o', callbackId, message.args);
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
       // @ts-ignore
       callback.apply(null, [...message.args, message.isPartialResponse]);
 
       // Remove the callback to ensure that the callback is called only once and to free up memory if response is a complete response
       if (!isPartialResponse(evt)) {
-        logger('Removing registered callback for message %i', message.id);
-        delete CommunicationPrivate.callbacks[message.id];
+        logger('Removing registered callback for message %i', callbackId);
+        CommunicationPrivate.callbacks.delete(callbackId);
```

Comment:
nit: I wonder if it would be helpful to take lines 679 -681 and turn them into a helper function that looked something like:

```typescript
function removeMessageHandlers(message: MessageResponse, map: Map<MessageUUID, unknown>): void {
  const callbackId = retrieveMessageUUIDFromResponse(message);
  map.delete(callbackId); // need to make sure delete works since this might have by-value problems
  if (!message.uuid) {
    delete CommunicationPrivate.legacyMessageIdsToUuidMap[message.id];
  }
}
```